src/stream/src/executor/managed_state/dynamic_filter.rs

```diff
@@ -415,8 +449,44 @@ fn range_contains_lower_upper(
 
 #[cfg(test)]
 mod tests {
+    use risingwave_common::catalog::{ColumnDesc, ColumnId, TableId};
+    use risingwave_common::types::DataType;
+    use risingwave_common::util::sort_util::OrderType;
+    use risingwave_storage::memory::MemoryStateStore;
+    use risingwave_storage::table::Distribution;
+
     use super::*;
 
+    #[tokio::test]
+    async fn test_shrink() {
+        let fallback = Distribution::fallback();
+        let mem_state = MemoryStateStore::new();
+        let column_descs = ColumnDesc::unnamed(ColumnId::new(0), DataType::Int64);
+        let state_table = StateTable::new_without_distribution(
+            mem_state.clone(),
+            TableId::new(0),
+            vec![column_descs.clone()],
+            vec![OrderType::Ascending],
+            vec![0],
+        )
+        .await;
+        let mut range_cache = RangeCache::new(state_table, usize::MAX, fallback.vnodes);
+        range_cache.init(EpochPair::new_test_epoch(1));
+        range_cache
+            .range((Bound::Unbounded, Bound::Unbounded), false)
+            .await
+            .unwrap();
+        assert_eq!(
+            range_cache.range,
+            Some((Bound::Unbounded, Bound::Unbounded))
+        );
+        range_cache.shrink(&ScalarImpl::Int64(5));
+        assert_eq!(
+            range_cache.range,
+            Some((Bound::Included(ScalarImpl::Int64(5)), Bound::Unbounded))
```

Comment:
Can we be slightly more comprehensive covering the behaviours?

Reply:
> Can we be slightly more comprehensive covering the behaviours?

fixed

Reply:
Thanks, much better.